back-end/scripts/importAuthors.py

```diff
@@ -0,0 +1,113 @@
+import json
+import mysql.connector
+from mysql.connector import errorcode
+import requests
+
+CONFIG = {
+  'user': 'root',
+  'password': 'root',
+  'host': '127.0.0.1',
+  'database': 'huskyreads',
+  'port': 2000,     # MAKE SURE THIS PORT ALIGNS WITH YOUR PORT IN THE .env FILE
+  'raise_on_warnings': True
+}
+
+def connectToDatabase():
+    """ Creates connection to the MySQL Database
+
+    Returns:
+        The database connection object
+    """
+    try:
+        cnx = mysql.connector.connect(**CONFIG)
+        return cnx
+    except mysql.connector.Error as err:
+        if err.errno == errorcode.ER_ACCESS_DENIED_ERROR:
+            print("Something is wrong with your user name or password")
+        elif err.errno == errorcode.ER_BAD_DB_ERROR:
+            print("Database does not exist")
+        else:
+            print(err)
+
+def retrieveAuthorIds(filePath: str):
+    """ Retrieves unique author ids for authors who wrote the books in the provided file
+
+        Args:
+            filePath:
+                The file-path to the JSON file containing the data
+
+        Expected:
+            JSON file to contain field "data", with value [arr] where each indice contains book data
+
+        Returns:
+            Set of author ids
+    """
+    f = open(filePath)
+    data = json.load(f)
+    # Assumes data field is called "data" - this is controlled by the data modifying code
+    authorIds = set()
+    for book in data["data"]:
+        authors = book.get("authors")
```

Comment:
Put the code with `get()` function calls under try-except where it catches KeyError in case any of these keys don't exist. Just keeps the code more robust.